server.js

```diff
@@ -15,10 +17,81 @@ const app = express();
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: "Service unavailable" });
+  }
+});
+
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    unique: true,
+    minlength: [5, "The message must be at least 5 characters"],
+    maxlength: [140, "The message must be less than 140 characters"],
+    trim: true,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date(),
+  },
+});
+
+const Thought = mongoose.model("Thought", ThoughtSchema);
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+app.get("/thoughts", async (req, res) => {
+  try {
+    const thoughts = await Thought.find({})
+      .sort({ createdAt: "desc" })
+      .limit(20)
+      .exec();
+    res.status(201).json(thoughts);
```

Comment:
This is a small and totally optional thing, but if you would like to have your code be as symmetrical as possible in the backend, you could implement the response here in similar style as you do in the error response, having an object with two properties, one for success: true and the other showing the data,  in a way you want to name it (response: thoughts/data: thoughts). I also noticed that you have used this in other paths, with response so maybe it will also be more unified to use the same approach throughout the code 😊